mis_builder/models/aep.py

```
@@ -238,7 +241,7 @@ def get_aml_domain_for_dates(self, date_from, date_to,
         elif mode == self.MODE_UNALLOCATED:
             date_from_date = fields.Date.from_string(date_from)
             fy_date_from = \
-                self.company.\
+                self.company_ids[0].\
```

Hm. Here it's a special treatment for the first company. This could lead to surprises. 

Could you extract the `fy_date_from` computation in a separate method and add a check (raise UserError) that all companies have the same `fy_date_from`? Later we can add logic to handle companies with different fiscal years.

---

I can see some border effects when company is located in Australia with GMT +12, but this can be fixed.
However don't you think that for this particular case (at least) the main company (or user.company_id) is useful?

---

It could be useful to have a main company but not in AEP. Perhaps in mis_report_instance. But even that I'm not sure and I certainly have not studied the impact of having different fiscal years on the computation of consolidated initial balances... Until that is clarified I prefer to raise an error if the various companies have different fiscal year dates.